tests/debugpy/test_evaluate.py

```diff
@@ -677,3 +677,88 @@ def code_to_debug():
 
         session.request_continue()
         assert backchannel.receive() == 1000
+
+
+def test_evaluate_thread_locks(pyfile, target, run):
+    @pyfile
+    def code_to_debug():
+        """
+        The idea here is that a secondary thread does the processing of instructions,
+        so, when all threads are stopped, doing an evaluation for:
+
+        processor.process('xxx')
+
+        would be locked until secondary threads start running.
+        See: https://github.com/microsoft/debugpy/issues/157
+        """
+
+        import threading
+
+        try:
+            from queue import Queue
+        except:
+            from Queue import Queue
+
+        class EchoThread(threading.Thread):
```

Comment:
Tests get a handy helper to reduce verbiage like this - take a look at the use of `@daemon` in `test_timeline.py`. Aside from conciseness, it also checks that the thread doesn't outlive the test in which it was spawned.

Reply:
Oh, nevermind, I missed the fact that this is inside debuggee, so you don't get the fixtures.